docs/src/components/ComponentPage/ComponentPage.js

```diff
@@ -0,0 +1,186 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow strict */
+
+import React, { type Node } from 'react';
+import { cssModules } from 'bpk-react-utils';
+import BpkContentContainer from 'bpk-component-content-container';
+import { BpkList, BpkListItem } from 'bpk-component-list';
+import BpkLink from 'bpk-component-link';
+import BpkText, { TEXT_STYLES, WEIGHT_STYLES } from 'bpk-component-text';
+
+import BpkMarkdownRenderer from '../DocsPageBuilder/BpkMarkdownRenderer';
+import getMarkdownString from '../../helpers/markdown-helper';
+import UsageTable from '../UsageTable';
+
+import STYLES from './ComponentPage.scss';
+
+const getClassName = cssModules(STYLES);
+
+type ContentType = string | Node;
+
+type Props = {
+  examples: Array<{
+    id: string,
+    title: string,
+    blurb?: ContentType,
+    content?: ContentType,
+  }>,
+  readme: string,
+  additionalContent: ?Array<{
+    id: string,
+    title: string,
+    content: ContentType,
+  }>,
+  usageTable: ?{
```

Comment:
Doesnt' this require us to explicitly pass `null` and thus isn't the type

```js
usageTable?: { 
  dos: Array<string>,
  donts: Array<string>.
}
```

better?

Reply:
Yeah you're right, will change.

Reply:
Ah I just remembered why I did it this way 😓 

<img width="492" alt="Screen Shot 2020-08-25 at 17 29 18" src="https://user-images.githubusercontent.com/73652/91201576-83754500-e6f8-11ea-8f6f-2e32eba8358f.png">

Obviously I can add the `defaultProps` declaration, but what's the point if the former way works?

Reply:
My understanding was that this is how you are recommended to do it. But here's what the docs [says](https://flow.org/en/docs/react/components/#toc-using-default-props)

Reply:
Ah interesting. Tbh I don't have a strong opinion on this. My philosophy with Flow is "whatever it takes to get it to stop complaining". So I can either change it and add `defaultProps`, or leave it as is. 

I lean towards leaving it, because it's less code, but as I say, I don't mind. What do you think?